### ISSUE TITLE
Support loading screen while assets download and entities load

### ISSUE THREAD

**Issue by cvan** [aframevr/aframe-core#718]
Per https://twitter.com/footedesign/status/677967278890221570

**Comment by ngokevin** [comment]
depends on adding asset blocking to `<a-assets>` before it fires loaded.

then users than listen on `loaded` and manage loading screens however they want.

**Comment by cvan** [comment]
@ngokevin but in the case of external URLs that get loaded from `material` component and then from `THREE.js`'s loaders, we can handle those today without changing making `<a-assets>` inert (which we could do, but then we'd have to enable some like `preload` boolean).

**Comment by ngokevin** [comment]
i think we intend if you don't put it in assets there is no preloading/blocking logic

**Comment by cvan** [comment]
> i think we intend if you don't put it in assets there is no preloading/blocking logic.

yeah, I know - basically what this issue is for is to introduce some type of events of `assetsstart`, assetsprogress`, and `assetsend` or something for materials on entities or models to know when the asset(s) have been actually loaded.

**Comment by ngokevin** [comment]
Yeah we'd add events. But wouldn't have to touch the material component itself

**Comment by cvan** [comment]
> Yeah we'd add events. But wouldn't have to touch the material component itself

correct, just the `loadImage`, `loadVideo`, etc. functions in that file
